### PR TITLE
Manual top-level operators

### PR DESCRIPTION
**Pull request by konnov**
I have started to write Part 2 of the language manual. To avoid huge PRs this time, this PR contains material on the overview of user-defined operators and details on top-level user-defined operators. The other pages are just stubs.

- [X] Entry added to [CHANGES.md](./CHANGES.md) for any new functionality

Pages in readable form can be found here:
 - [User operators](https://github.com/informalsystems/apalache/blob/igor/lang2/docs/src/lang/user-operators.md)
 - [Top-level operators](https://github.com/informalsystems/apalache/blob/igor/lang2/docs/src/lang/top-level-operators.md)

### CODE REVIEW

**Review by shonfeder — CHANGES_REQUESTED** [review]
Lol. Forgot to hit submit on these comments last week! Sorry :/

File: docs/src/lang/lambdas.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Anonymous operator definitions
+
+work in progress...
+
+[[Back to user operators]](./user-operators.md)
```
[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/lambdas.md:5]: They are all related to user operators. Good idea.

File: docs/src/lang/user-operators.md [hunk 2]
```diff
@@ -1,12 +1,103 @@
 # User-defined operators and recursive functions
 
-_Like macros, to do a lot of things in one step_
+_Like macros, to do a lot of things in one system step..._
 
- - Top-level operators
+## For the impatient
```
[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/user-operators.md:5]: I am trying to achieve two things at once here: Give the reference material to those who know the language already, and give an introduction to the concepts to those who don't know it yet. So "the impatient" section is for those who want to refresh their memory without reading too much introductory text.

File: docs/src/lang/lambdas.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/lambdas.md:5]: Fixed

File: docs/src/lang/user-operators.md [hunk 3]
```diff
@@ -1,12 +1,103 @@
...
+
+Recently, Leslie Lamport has extended the syntax of TLA+ operators in [TLA+
+version 2], which supports recursive operators and lambda operators.  The
+operator syntax that is described in [Specifying Systems] describes TLA+
+version 1. This page summarizes the syntax of user-defined operators in
+versions 1 and 2.
+
+**Short digression**. The most important thing to understand about operators is
+that they are normally used inside `Init` and `Next` that do not describe the
+whole system computation! The operator `Next` describes a single step of the
```
[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/user-operators.md]: Fixed

File: docs/src/lang/user-operators.md [hunk 4]
```diff
@@ -1,12 +1,103 @@
...
+  1. Operators do not support [Currying]. That is, you can only apply an operator
+  by providing values for all of its expected arguments.
+
+  1. Operators can be nested. However, nested operators require a slightly
+  different syntax. They are defined with LET-IN definitions.
+
+**Details about operators.** We go in detail about different kinds of operators
+and recursive functions below:
+
+ - [Top-level operator definitions](./top-level-operators.md)
```
[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/user-operators.md]: Fixed

File: docs/src/lang/user-operators.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (shonfeder) [reply docs/src/lang/user-operators.md:5]: I'd content that, with the table of contents now available in the side bar, you don't need this redundant listings We know have the same items listed out in triplicate. Those who just need quick reference will have it at their fingertips in the sidebar nav.

[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/user-operators.md:5]: That makes sense. I will remove that section

File: docs/src/lang/top-level-operators.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+expansion], though it is a bit smarter: It does not blindly mix names from the
+operator's body and its application context. For example, the following
+semantics by substitution is implemented in the [Apalache] model checker.
+
+Here we give a simple explanation for non-recursive operators. Consider the
+definition of an `n`-ary operator `A` and its application in the definition
+of another operator `B`:
+
+```tla
+A(p_1, ..., p_n) == body_of_A
```
[the remark this answers is not in the document]
Comment (konnov) [reply docs/src/lang/top-level-operators.md]: Right. I have renamed the parameters of `B` to `p_1, ..., p_k`, to make it more obvious.

**Review by shonfeder — APPROVED** [review]
LGTM

**Pull request merged**